torch_glow/src/PyTorchLoaderTest.cpp

```diff
@@ -16,149 +16,45 @@
-      torch::jit::lastExecutedOptimizedGraph();
 
-  glow::Module mod;
-  auto *f = mod.createFunction("PyTorchFunction");
-  std::vector<glow::Placeholder *> inputPlaceholders;
-  std::vector<glow::Placeholder *> outputPlaceholders;
-
-  // Add self parameter.
-  vec.insert(vec.begin(), module->module_object());
-
-  at::ArrayRef<torch::jit::IValue> inputs(vec);
-  errPtr = glow::PyTorchModelLoader::loadJITGraph(
-      *f, *subgraph, inputs, inputPlaceholders, outputPlaceholders);
-  EXPECT_FALSE(!errPtr); // TODO
+  llvm::Error err = glow::PyTorchFileLoader::loadPyTorchModel(fileName, module);
+  assert(!err);
 }
 
-TEST(PyTorchModelLoader, Fuser) {
+void pyTorchModelLoaderFusionTest() {
   const std::string fileName{GLOW_DATA_PATH
                              "tests/models/pytorchModels/resnet18.pt"};
-  std::shared_ptr<torch::jit::script::Module> module;
-  llvm::Error errPtr = llvm::Error::success();
-  glow::PyTorchFileLoader fileLoader(fileName, module, &errPtr);
-  EXPECT_FALSE(errPtr);
-  std::shared_ptr<torch::jit::Graph> graph;
-  static auto testSymbol = at::Symbol::fromQualString("glow::LoaderTest");
 
   glow::Module mod;
-  localFunction = mod.createFunction("GlowFunction");
-
-  LOG(INFO) << "Register operator and pass";
-  {
-    auto options = c10::OperatorOptions();
-    options.setAliasAnalysis(at::AliasAnalysisKind::PURE);
-    torch::jit::RegisterOperators op({torch::jit::Operator(
-        testSymbol,
-        [](const torch::jit::Node *node) {
-          return [node](torch::jit::Stack &stack) {
-            // Get JIT Graph.
-            auto graph = node->g(at::attr::Subgraph);
-            convertJitGraphToGlowFunction(stack, *graph, *localFunction);
-            return 0;
-          };
-        },
-        options)});
-
-    torch::jit::RegisterPass pass([&](std::shared_ptr<torch::jit::Graph> &g) {
-      torch::jit::CustomFuseGraph(g, glow::PyTorchModelLoader::isNodeSupported,
-                                  testSymbol);
-    });
-  }
-
-  LOG(INFO) << "Running forward method...";
+  auto *F = mod.createFunction("GlowFunction");
   std::vector<torch::jit::IValue> vec;
   auto emptyTensor = at::empty({1, 3, 224, 224});
   vec.push_back(torch::autograd::make_variable(emptyTensor));
 
-  module->forward(vec);
-
-  // Find all Save nodes and their placeholders.
   std::vector<glow::Placeholder *> inputPlaceholders;
   std::vector<glow::Placeholder *> outputPlaceholders;
-  findInputAndOutputPlaceholders(*localFunction, inputPlaceholders,
-                                 outputPlaceholders);
 
-  EXPECT_EQ(inputPlaceholders.size(), 1);
-  localFunction = nullptr;
+  llvm::Error err = glow::PyTorchFileLoader::loadPyTorchGraph(
+      fileName, vec, *F, inputPlaceholders, outputPlaceholders);
+
+  if (static_cast<bool>(err)) {
+    LOG(ERROR) << llvm::toString(std::move(err)) << "\n"; // size and reshape.
+  }
+}
+} // namespace
+
+int main(int /*argc*/, char ** /*argv*/) {
```

Comment:
I wan't able to build gtest under github/CMakefile.

Reply:
Hmm ok well all of our other c++ tests use gtest with cmake, I think we should continue using it here

Reply:
I have meant:
python setup.py test --run_cmake
failed.

Reply:
What's the error?

Reply:
Something like, cannot find file: #include "gtest/gtest.h"

Reply:
And you included gtest in target_link_libraries?

Reply:
yes, I have added gtest to CMakefile.

Reply:
😞 oh well let's move no as is then.
It would be great if this was like the rest of our tests. Integration with pytorch jit via C++  is important and we should add some more tests here eventaully, we can follow up on trying to get gtest working then